VideoLocker/src/main/java/org/edx/mobile/view/DownloadListActivity.java

```diff
@@ -152,26 +152,32 @@ public DownloadItem(@NonNull DownloadEntry model, @NonNull NativeDownloadModel n
         }
 
         @Override
-        public long getSize() {
-            if (model.size == 0) {
+        @Nullable
+        public Long getSizeInBytes() {
+            if (model.size <= 0) {
+                if (nativeModel.size <= 0) {
+                    return null;
```

Comment:
I'd rather hide the implementation details of DownloadManager from the adapter; the adapter only cares "do we know the size or not?" and I think a `@Nullable` has better semantics for that. This interface exists solely for the adapter so it should hide as many low-level details as possible.

Having said that, the interface was exposing the DownloadManager status which I think was wrong, so I made a nice enum instead.